cloud/passportauth/src/auth/PassportAuth.ts

```diff
@@ -1,54 +1,51 @@
 import * as express from 'express';
 import * as passport from 'passport';
-import BaseUser from '../user/BaseUser';
-import UserSecService from '../user/UserSec';
+import {BaseUser} from '../user/BaseUser';
+import {UserSecService} from '../user/UserSec';
 
 interface Auth {
-  protect(req: express.Request, res: express.Response, next: () => void): void; // protect/authenticate?
-  login(opts?: object): void;
+  login(strategy: string, opts?: object): void;
```

Comment:
This is just passport.authenticate(). It doesn't handle any role checks or if user is already authenticated. It will just authenticate the user. The reason why it's called login here is because the equivalent of this is login() in keycloak @TommyJ1994